dashboard/api.py

```diff
@@ -132,13 +132,21 @@ def get_info_for_course(user, course, user_enrollments, user_certificates):
         return format_course_for_dashboard(run_status.course_run, CourseStatus.CURRENT_GRADE, course)
     # check if we need to check the certificate
     elif run_status.status == CourseRunStatus.READ_CERT:
-        # TODO: here should go the logic to check the actual certificate  # pylint: disable=fixme
-        #       for now just assume not passed and return the next available course run
-        # this is the case when the certificate says the user did not pass the course
+        # if there is no certificate for the user, the user never passed
+        # the course, so she needs to enroll in the next one
+        if not user_certificates.has_verified_cert(run_status.course_run.edx_course_key):
+            return format_course_for_dashboard(
+                course.get_next_run(),
+                CourseStatus.OFFERED if course.get_next_run() is not None else CourseStatus.NOT_OFFERED,
+                course
+            )
+        # pull the verified certificate for course
+        cert = user_certificates.get_verified_cert(run_status.course_run.edx_course_key)
```

Comment:
from my yesterday's conversation with Matt Dryer, no she does not get it. There is a special case when the instructor can make the student pass the course anyway, but in that case the certificate should be similar to a student that passed normally